### DISCUSSION TITLE
Symfony Twig - how to connect well with JS

### DISCUSSION THREAD

**Discussion by ciastko1742** [symfony/symfony#48175] [Q&A]
Hello,
I'd like to rewrite an old and fairly large Symfony application. However, at the beginning I have many problems to solve so that later they will not be there. One of them is creating the appearance of the application. Access to individual users with different views plays a big role in this application. with Twig I can add 'if role ==' with views in templates. And I don't have much experience with frameworks like React.js and writing pure REST applications to separate this.
The old app uses a lot of jQuery and there was a lot of user interaction. There is so much of it that there is a terrible confusion in the files. I would not like to do it as a SPA application, incl. to do REST in Symfony and frontend, among others in Vue.js, but there will be a lot of such elements from JS, incl. interactive calendar, AJAX search, etc. How do you go about it? What to use so as not to regret later? How can I add VUE.js components only for the calendar, and for the search engine another component that I will use only in the appropriate views?

**Comment by MatTheCat** [comment]
Hi,

I personally never tested it yet but I think you’ll be interested in https://ux.symfony.com/
